src/DropdownMenu.js

```diff
@@ -109,23 +98,23 @@ class DropdownMenu extends React.Component {
   }
 
   handleClose = e => {
-    if (!this.props.onToggle) return;
+    if (!this.context.onToggle) return;
 
-    this.props.onToggle(false, e);
+    this.context.onToggle(false, e);
   };
 
   render() {
-    const {
-      show,
-      flip,
-      menuRef,
-      alignEnd,
-      drop,
-      usePopper,
-      toggleNode,
-      rootCloseEvent,
-      popperConfig = {},
-    } = this.props;
+    const { flip, usePopper, rootCloseEvent, popperConfig = {} } = this.props;
+
+    const { drop, menuRef, toggleNode } = this.context;
+
+    const show =
+      this.context.show == null ? this.props.show : this.context.show;
+
+    const alignEnd =
+      this.context.alignEnd == null
```

Comment:
shouldn't props win over context in general?

Reply:
probably...this was the way it was i think